client_test.go

```diff
@@ -37,15 +41,88 @@ func TestTriggerSuccessCase(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+// Tests that when the "info" param is not specified, we get a nil Channels map in the returned TriggerChannelsList
+func TestTriggerWithParamsSuccessCase(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(200)
+		testJSON := "{}"
+		fmt.Fprintf(res, testJSON)
+		assert.Equal(t, "POST", req.Method)
+
+		expectedBody := map[string]interface{}{"name": "test", "channels": []interface{}{"test_channel"}, "data": "yolo"}
+		bodyDeoder := json.NewDecoder(req.Body)
+		var actualBody map[string]interface{}
+		err := bodyDeoder.Decode(&actualBody)
+		assert.NoError(t, err)
+		assert.Equal(t, expectedBody, actualBody)
+
+		assert.Equal(t, "application/json", req.Header["Content-Type"][0])
+		lib := fmt.Sprintf("%s %s", libraryName, libraryVersion)
+		assert.Equal(t, lib, req.Header["X-Pusher-Library"][0])
+		assert.NoError(t, err)
+	}))
+	defer server.Close()
+
+	u, _ := url.Parse(server.URL)
+	client := Client{AppID: "id", Key: "key", Secret: "secret", Host: u.Host}
+	// Empty parameters
+	channels, err := client.TriggerWithParams("test_channel", "test", "yolo", TriggerParams{})
+	assert.NoError(t, err)
+
+	expected := &TriggerChannelsList{
+		Channels: nil,
+	}
+	assert.Equal(t, expected, channels)
+}
+
+func TestTriggerWithParamsInfoSuccessCase(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(200)
+		testJSON := "{\"channels\":{\"test_channel\":{\"subscription_count\":1}}}"
```

Comment:
Nitpick: if you use backticks for these json strings, you won't need to escape the `"`

Reply:
I'll leave this for now since it's like this in all the tests, but might do in a follow up PR.